From 292e28ae1df4c5d138a93c037484c6435148f5cd Mon Sep 17 00:00:00 2001
Message-Id: <292e28ae1df4c5d138a93c037484c6435148f5cd.1417780177.git.chang-joon.lee@intel.com>
In-Reply-To: <e293659a31a2970805251d9fed030ac809a9a6d0.1417780177.git.chang-joon.lee@intel.com>
References: <e293659a31a2970805251d9fed030ac809a9a6d0.1417780177.git.chang-joon.lee@intel.com>
From: Adam Rutkowski <adam.j.rutkowski@intel.com>
Date: Sat, 23 Aug 2014 11:25:59 -0400
Subject: [PATCH 23/30] REVERTME [VPG]: drm/i915: Perfmon IOCTL v5 stubs

Adding stubs for Perfmon IOCTL v5 entry points.

Patch is marked with REVERTME as Perfmon IOCTL will be removed
once OA couters are exposed via 'perf event' framework.

Issue: GMINL-3120
Change-Id: Ieabecb16b9206ec8b369d42d40719282512897af
Signed-off-by: Adam Rutkowski <adam.j.rutkowski@intel.com>
---
 drivers/gpu/drm/i915/i915_perfmon.c |   48 +++++++++++++++++++++++++++++++----
 1 file changed, 43 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_perfmon.c b/drivers/gpu/drm/i915/i915_perfmon.c
index 807d431..01fb04c 100644
--- a/drivers/gpu/drm/i915/i915_perfmon.c
+++ b/drivers/gpu/drm/i915/i915_perfmon.c
@@ -97,29 +97,67 @@ int i915_perfmon_ioctl(struct drm_device *dev, void *data,
 	struct drm_file *file)
 {
 	struct drm_i915_perfmon *perfmon = data;
-	int retcode = 0;
+	int ret = 0;
 
 	switch (perfmon->op) {
 	case I915_PERFMON_SET_BUFFER_IRQS:
-		retcode = intel_enable_perfmon_interrupt(
+		ret = intel_enable_perfmon_interrupt(
 				dev,
 				perfmon->data.set_irqs.enable);
 		break;
 	case I915_PERFMON_WAIT_BUFFER_IRQS:
 		if (perfmon->data.wait_irqs.timeout >
 				I915_PERFMON_WAIT_IRQ_MAX_TIMEOUT_MS)
-			retcode =  -EINVAL;
+			ret =  -EINVAL;
 		else
 			perfmon->data.wait_irqs.ret_code =
 				intel_wait_perfmon_interrupt(
 					dev,
 					perfmon->data.wait_irqs.timeout);
 		break;
+
+	case I915_PERFMON_CANCEL_WAIT_BUFFER_IRQS:
+		ret = -ENODEV;
+		break;
+
+	case I915_PERFMON_OPEN:
+		ret = -ENODEV;
+		break;
+
+	case I915_PERFMON_CLOSE:
+		ret = -ENODEV;
+		break;
+
+	case I915_PERFMON_ENABLE_CONFIG:
+		ret = -ENODEV;
+		break;
+
+	case I915_PERFMON_DISABLE_CONFIG:
+		ret = -ENODEV;
+		break;
+
+	case I915_PERFMON_SET_CONFIG:
+		ret = -ENODEV;
+		break;
+
+	case I915_PERFMON_LOAD_CONFIG:
+		ret = -ENODEV;
+		break;
+
+	case I915_PERFMON_GET_HW_CTX_ID:
+		ret = -ENODEV;
+		break;
+
+	case I915_PERFMON_GET_HW_CTX_IDS:
+		ret = -ENODEV;
+		break;
+
 	default:
+		DRM_DEBUG("UNKNOWN OP\n");
 		/* unknown operation */
-		retcode = -EINVAL;
+		ret = -EINVAL;
 		break;
 	}
 
-	return retcode;
+	return ret;
 }
-- 
1.7.9.5

